os/kernel/group/group_leave.c

```diff
@@ -414,26 +387,61 @@ void group_leave(FAR struct tcb_s *tcb)
 		/* Have all of the members left the group? */
 
 		if (group->tg_nmembers == 0) {
-#ifdef CONFIG_APP_BINARY_SEPARATION
-			if (IS_LOADED_MODULE(group)) {
-				/* The region of stack is allocated in user RAM partition.
-				 * If it is loaded, whole RAM partition for binary is freed by unloading in group_release.
-				 * So the stack doesn't need to freed after unloading.
-				 */
-
-				tcb->stack_alloc_ptr = NULL;
-			}
-#endif
+
 			/* Yes.. Release all of the resource held by the task group */
 
+#ifdef CONFIG_BINFMT_LOADABLE
+			if ((tcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_TASK) {
+				struct task_tcb_s *task_tcb = (struct task_tcb_s *)tcb;
+				if (IS_BINARY_MAINTASK(task_tcb)) {
+					/* The region of stack is allocated in user RAM partition.
+					 * If it is loaded, whole RAM partition for binary is freed by unloading in group_release.
+					 * So the stack doesn't need to freed after unloading.
```

Comment:
I agreed that unloading threads' stack don't need to be freed. But current implementation, we skip the stack release only for main task. How about skipping all of the member threads also?

Reply:
I raised the PR about skipping member thread stack release. Please check this.
https://github.com/Samsung/TizenRT/pull/4576